server/cast/char.go

```diff
@@ -146,6 +146,20 @@ func charExplicit() {
 			return d, nil
 		},
 	})
+	framework.MustAddExplicitTypeCast(framework.TypeCast{
+		FromType: pgtypes.BpChar,
+		ToType:   pgtypes.Oid,
+		Function: func(ctx framework.Context, val any, targetType pgtypes.DoltgresType) (any, error) {
+			out, err := strconv.ParseInt(strings.TrimSpace(val.(string)), 10, 32)
+			if err != nil {
+				return nil, fmt.Errorf("invalid input syntax for type %s: %q", targetType.String(), val.(string))
+			}
+			if out > 2147483647 || out < 0 {
```

Comment:
`uint32` goes beyond `2147483647`, so we should update this. We should also add tests for this (at max/min and max+1 and min-1).